### ISSUE TITLE
Restructure

### ISSUE THREAD

**Issue by bambooforest** [cldf-datasets/bdproto#4]
From the email thread:

```
So in the `values.csv` table the columns `SourceLanguageName` and 
`SourceLanguageFamily` seem to be solely dependent on the `Inventory_ID` 
column, which is not a primary key (or key candidate on its own).  Thus 
it would be more sensible to put these into `contributions.csv`.

As mentioned before, the `Value` column seems to be dependent on 
`Parameter_ID`, too, and strictly speaking `Language_ID` has the same 
transitive dependency on `Inventory_ID`. But eliminating these columns 
would conflict with the cldf standard, as far as i know.

Also in `contributions.csv` there are dependent on the language or maybe 
glottocode in some way (e.g. again there is a column for `LanguageName` 
and `LanguageFamily`; I'm not sure about others like `Homeland` or 
`Type`).  So maybe these should be moved to `language.csv`.
```

And

```
Ah, ok. I'm sympathetic to moving SourceLanguageName/Family to 
contributions.csv, because that's where dataset-specific metadata for 
the varieties under investigation should go.

The Value column in values.csv mirroring Parameter_ID is due to the 
nature of "language has phoneme" parameters: The presence of a row in 
values.csv already captures this semantic, so we are free to push 
something into Value which makes re-use easier.

If LanguageName or LanguageFamily in contributions.csv is derived from 
Glottolog, I agree that these two fields should go into languages.csv.
```

So, we should:

- move SourceLanguageName/Family to contributions.csv

We supply LanguageName and LanguageFamily due to the lack of glottocodes for some of these ancient or reconstructed languages.

@xrotwang -- thoughts?

**Comment by blurks** [comment]
I'm still quite new to the cldf standard and might miss some practical considerations or conventions.  So please correct me, if I'm wrong or overlook something.

But from a technical point of view, I would suggest to remove `LanguageName` and `LanguageFamily` from `contributions.csv` and use the corresponding fields  in `language.csv` instead.  A row in `languages.csv` doesn't necessarily need a glottocode.  As for now, glottocodes are used as `Language_ID` inside `languages.csv`, but actually you can put basically anything you want as inside `Language_ID`.  E.g. note that [in this table](https://github.com/cldf-datasets/wals/blob/master/cldf/languages.csv) which is referenced as an example in the [cldf specification](https://github.com/cldf/cldf/tree/master/components/languages), the first row doesn't have a glottocode.

If all languages of the dataset would have a unique `Language_ID` in `languages.csv`, regardless of them being included in glottolog or not, one could also reference them from `values.csv`.  Right now some of the rows in `languages.csv` are missing a `Language_ID`, which conflicts with the cldf standard, as far as I can tell.  At least validating the dataset gives a lot of warnings about that:
```sh
$ cldf validate cldf/StructureDataset-metadata.json 
WARNING cldf/values.csv:72:2 Language_ID: required column value is missing
WARNING cldf/values.csv:73:2 Language_ID: required column value is missing
WARNING cldf/values.csv:74:2 Language_ID: required column value is missing
WARNING cldf/values.csv:75:2 Language_ID: required column value is missing
WARNING cldf/values.csv:76:2 Language_ID: required column value is missing
WARNING cldf/values.csv:77:2 Language_ID: required column value is missing
...
```

There might be an argument, though, for separating data (like language names) derived from glottolog from other data (eg. derived from the sources).  I believe this is what @xrotwang  meant at the end of your citations from the email thread.  My take on that would be that the presence of a glottocode in a row of `languages.csv` labels that row as being "authorized" by the glottolog, while the absence of a glottocode marks its somewhat precarious status.  But as I am neither a linguist nor a researcher I don't know if that's a valid approach.

**Comment by blurks** [comment]
So we also found some conflicts between the language names in  `contributions.csv` and `languages.csv`

e.g. there are these rows  in `contributions.csv`:

| ID | SpecificDialect | LanguageName       | LanguageFamily | LanguageFamilyRoot | Glottocode |
| -- | --------------- | ------------------ | -------------- | ------------------ | ---------- |
|  3 |                 | Proto-Balto-Slavic | Balto-Slavic   | Indo-European      | balt1263   |
| 32 |                 | Proto-Balto-Slavic | Balto-Slavic   | Indo-European      | balt1263   |
| 36 |                 | Proto-Baltic       | Balto-Slavic   | Indo-European      | balt1263   |

As you can see, these three share the same glottocode but have different names.  The name associated with this glottocode in `languages.csv` is yet another one:

| ID       | Name         | Macroarea | Latitude | Longitude | Glottocode |
| -------- | ------------ | --------- | -------- | --------- | ---------- |
| balt1263 | Balto-Slavic | Eurasia   |          |           | balt1263   |

I think these conflicts would be best resolved by having language names only in `languages.csv` and by then referring to languages in other tables only by there respective `LanguageID` (which doesn't have to be a glottcode, as mentioned in my previous comment).  This is also the way [phoible does it](https://github.com/cldf-datasets/phoible/tree/master/cldf#table-inventoriescsv).  Note that in the PHOIBLE cldf, `inventories.csv` (which is basically our `contributions.csv`) doesn't hold any data about languages.  Instead associating languages with contributions is done via `values.csv`.  This association is implied already by the structure of `values.csv`, so we wouldn't even need to restructure that one.  I don't know if that's the best way to model it, but  "since BDPROTO data is basically just PHOIBLE data" as was said via email, it might make interoperability between the two datasets much more easy, if they have the same (or similar) structure.

So to achieve a PHOIBLE-like structure and to resolve the conflicts mentioned above, I'd add the following two steps:
* move all data on languages, which is not specific to a single contribution to `languages.csv` (i.e.  LanguageName, LanguageFamily, LanguageFamilyRoot, Macroarea and probably also Homeland)
* make all Languages addressable via a LanguageID in `languages.csv` and associate languages, contributions and phonemes via `values.csv`.  This is already done for all languages which have a glottocode, as this serves as a LanguageID right now

**Comment by bambooforest** [comment]
I'm OK with this. @xrotwang ? 

@blurks -- We would then take your suggestions and ask @J22Melody to update the CLDF pipeline.

Regarding finishing the data referencing and corrections, @EitanGrossman is on that and has another week to complete them. Once finished, we'll run the final data through the pipeline with the new updates.

**Comment by blurks** [comment]
also don't forget to include your bibtex file containing the sources once it's done and to add a `dc:source` property to [`StructureDataset-metadata.json`](https://github.com/cldf-datasets/bdproto/blob/main/cldf/StructureDataset-metadata.json).   see the cldf spcec on the [sources file](https://github.com/cldf/cldf#sources-reference-file) for details

**Comment by bambooforest** [comment]
All references for inventories are now available here:

https://github.com/bdproto/bdproto/blob/master/sources.bib

**Comment by bambooforest** [comment]
Closed by #6
